### DISCUSSION TITLE
How to build a typescript library?

### DISCUSSION THREAD

**Discussion by rte76702** [parcel-bundler/parcel#9320] [General]
How do I build this sample library: https://github.com/rte76702/simple-ts-lib
so it can be used like this:
```ts
/* file_in_another_package_with_simple-ts-lib_installed_as_dependency.ts */

// this works:
import { operation } from "simple-ts-lib"

// this doesn't work:
import { add } from "simple-ts-lib/utils"

// this works, but i want build with js files in /dist
// and corresponding types.d.ts, in place of ts files in /src
import { add } from "simple-ts-lib/src/utils"
```

such that simple-ts-lib can be published with just the contents of /dist (without /src)

The example in the site here: https://parceljs.org/getting-started/library/ does not allow the use of other modules oustide index.ts (here utils.ts) by the library end user

**Comment by mischnic** [comment]
Use `includeNodeModules`: https://parceljs.org/features/targets/#includenodemodules

**Reply by rte76702** [reply]
Hi, thanks for the reply,
Please could you help with what exactly I am supposed to set that option to?
I have no idea what to do with that option and the linked documentation does not provide any insight in relation to what I am trying to achieve.

Please see a sample 'consumer' repo with the usage I am trying to achieve:
https://github.com/rte76702/sample-ts-lib-user/blob/17bbf1eb442dedf993329da014bf0b2a179ba9fa/index.ts#L3


Here is a codesandbox link showing the error on index.ts line 3 which I would like to get rid of:
https://codesandbox.io/p/github/rte76702/sample-ts-lib-user/main
